### ISSUE TITLE
The trained policy weights cannot accomplish the tasks

### ISSUE THREAD

**Issue by uenian33** [Cranial-XIX/metric-residual-network#1]
I used the run_all.sh scripts with "asym_max" critic that trained policies for tasks. However, when I tried loading the pretrained weights, none of the tasks can be accomplished. Here is the function I create for policy loading:

```
def load_model(self):
        #print("esults/HandManipulateBlockRotateZ_her_(-)rew_asym-max_emb16_lr0.001_sd100.pt")
        sd = torch.load( os.path.join(self.args.save_dir, f"{self.args.experiment_name}.pt"))
        self.actor.load_state_dict(sd["actor"])
        if hasattr(self, "critic"):
            self.critic.load_state_dict(sd["critic"])
        self.actor.eval()
        return
```

and I used the "eval_agent()" for success rate evaluation

**Comment by Cranial-XIX** [comment]
Hi,

Thanks for asking! I think you probably need to load the s_norm, s_std, g_norm, g_std (see [here](https://github.com/Cranial-XIX/metric-residual-network/blob/master/src/agent/base.py#L176)). The reason is that following the convention from previous work in HER, we normalize the state and goal to zero mean and 1 standard deviation, and the normalizer is formed online. So when you load the policy, you probably want to also update the normalizer. Please let me know if that still does not work :)

**Comment by uenian33** [comment]
Hi, Thanks for your answer! Now it works and sorry that I ignored the normalizers.

**Comment by Cranial-XIX** [comment]
Glad it works. Good luck with your research!
